packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1127,8 +1127,32 @@ abstract class State<T extends StatefulWidget> with Diagnosticable {
   /// ```
   ///
   /// It is an error to call this method after the framework calls [dispose].
-  /// You can determine whether it is legal to call this method by checking
+  /// You can determine whether it is legal to call [setState] by checking
   /// whether the [mounted] property is true.
+  ///
+  /// # Performance considerations
+  ///
+  /// Calls to [setState] cause a rebuild of the current widget and,
+  /// potentially, all of its children. This isn't efficient and generally not
+  /// desiderable. A [setState] call should only be made on widgets with small
+  /// subtrees in order to rebuild the least amount of children.
+  ///
+  /// A better alternative to [setState] is represented by the combined usage
+  /// of a [Listenable] along with an [AnimatedBuilder], which is:
```

Comment:
This focuses a bit too much on AnimatedBuilder. It isn't the only solution and we want to avoid that people overcorrect and think `setState` is always bad and must be replaced with AnimatedBuilders. AnimatedBuilder can be the right choice for some use cases. In other cases it's may be more practical for example to just push the state further down to the leaves. Maybe take a look at the "Performance considerations" section at https://master-api.flutter.dev/flutter/widgets/StatefulWidget-class.html and phrase this here a little more neutral stating in what situations setState is acceptable and when it may be problematic followed by some suggested alternatives that include more than just AnimatedBuilder.

Reply:
(or you could just link to the "Performance considerations" section at https://master-api.flutter.dev/flutter/widgets/StatefulWidget-class.html, which IMO covers this pretty well.)